app/models/order_details_updater.rb

```diff
@@ -84,4 +90,12 @@ def find_value(item_id)
     Item.where(id: params[:order][:order_details][:item_id]).find_each { |item| @cache[item.id] = item }
     @cache[item_id].value
   end
+
+  def deduct_from_inventory
+    order_details.each do |order_detail|
+      item = order_detail.item
+      item.current_quantity -= order_detail.quantity
```

Comment:
Please ensure there is a rails validation that will ensure the quantity cannot be less than 0. The validation would then cause the transaction to fail, which will be very important.